TypeCobol/Compiler/Diagnostics/TypeCobolChecker.cs

```diff
@@ -885,6 +886,44 @@ public static void CheckStatement(Node node)
                     }
                 }
             }
+
+            if (node.StorageAreaWritesDataDefinition != null && node.CodeElement is SetStatementForConditions setConditions && setConditions.IsSendingValueFalse)
+            {
+                // Statement is a SET TO FALSE: check whether it mixes variables of type BOOL and Level 88
+                bool hasBool = false;
+                bool hasLevel88 = false;
+                string errorMessage = "The\"SET\"statement mixes variables of type BOOL and Level 88. This is not possible: please split it into 2 statements.";
+                foreach (var condition in setConditions.Conditions)
+                {
+                    if (condition.StorageArea != null)
+                    {
+                        node.StorageAreaWritesDataDefinition.TryGetValue(condition.StorageArea, out var dataCondition);
```

Comment:
Same remark. Ideally we should check the return value. But as it is used that way in other places...

Reply:
Yes we did the mistake in other places, but it's better to write new code correctly.
So can you check the return value of `TryGetValue` ?

Reply:
Done